src/others/observable/lv_observable.h

```diff
@@ -0,0 +1,263 @@
+ */
+void lv_subject_set_ptr(lv_subject_t* subject, void* value);
+
+/**
+ * Update the value of a color type subject
+ * @param subject address of the subject
+ * @param color new value
+ * @note automatically notifies all observers
+ */
+void lv_subject_set_color(lv_subject_t* subject, lv_color_t color);
+
+/**
+ * Copy the string into the buffer of a string type subject
+ * @param subject address of the subject
+ * @param buf string to copy 
+ * @note automatically notifies all observers
+ */
+void lv_subject_copy_string(lv_subject_t* subject, char* buf);
+
+/**
+ * Set bitmask and value comparison pattern of the observer
+ * @param observer address of the observer
+ * @param pattern pattern used to mask out interested bits from the bitmask's value and compare it with a constant value
+ * @note The value
+ */
+void lv_observer_bitmask_set_cond(lv_observer_t* observer, lv_bitmask_pattern_t pattern);
+
+/**
+ * Bind bitmask to object flag
+ * @param subject address of the subject
+ * @param obj the LVGL object what will be affected by the change
+ * @param flag the affected flag
+ * @param pattern pattern used to mask out interested bits from the bitmask's value and compare it with a constant value
+ */
+void* lv_bind_bitmask_to_obj_flag(lv_subject_t* subject, lv_obj_t* obj, lv_obj_flag_t flag, lv_bitmask_pattern_t pattern);
+
+/**
+ * Bind bitmask to object state
+ * @param subject address of the subject
+ * @param obj the LVGL object what will be affected by the change
+ * @param state the affected state
+ * @param pattern pattern used to mask out interested bits from the bitmask's value and compare it with a constant value
+ */
+void* lv_bind_bitmask_to_obj_state(lv_subject_t* subject, lv_obj_t* obj, lv_state_t flag, lv_bitmask_pattern_t pattern);
+
+/**
+ * Bind int value to a local style property
+ * @param subject address of the subject
+ * @param obj the LVGL object what will be affected by the change
+ * @param prop the style property
+ * @param selector the part/state selector
+ */
+void* lv_bind_int_to_obj_local_style_prop(lv_subject_t* subject, lv_obj_t* obj, lv_style_prop_t prop, lv_style_selector_t selector);
+
+/**
+ * Bind string value to a label text
+ * @param subject address of the subject
+ * @param obj the LVGL object what will be affected by the change
+ */
+void* lv_bind_string_to_label_text(lv_subject_t* subject, lv_obj_t* obj);
+
+/**
+ * Bind int value to a label text using a predefined format string
+ * @param subject address of the subject
+ * @param obj the LVGL object what will be affected by the change
+ * @param fmt the format string
+ */
+void* lv_bind_int_to_label_text_fmt(lv_subject_t* subject, lv_obj_t* obj, const char* fmt);
+
+/**
+ * Bind and int value to a label text using a variable format string
+ * @param subject address of the subject
+ * @param fmt the address of the format string (an observable string value)
+ * @param obj the LVGL object what will be affected by the change
+ */
```

Comment:
Add
```c
@note if any of the subjects changes the label's text will be updated.
```

Reply:
Added:
@note If either the value or the formwat string changes the label's text will be updated.